### PR TITLE
an initial form generator

### PR DESCRIPTION
**Pull request by cflipse**
An initial construction of a form generator.  Comments welcome.

### CODE REVIEW

**Comment by cflipse** [comment]
There are a few slight changes from the forms used in the specs; I'm using the attributes hash in the update, and there are a few places where I suspect other input might be good.

requiring the command switch seems fairly un-railsy, but it'll do for a start.  I'm half-considering if it's a good idea to generate the base, extension pattern used in the spec/dummy app...

File: spec/lib/generators/form_generator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+              class NewUserForm < ROM::Model::Form
+                commands users: :create
+
+                input do
+                  set_model_name 'User'
+
+                  # define form input attributes
+                  # attribute :name, String
+
+                  timestamps :created_at, :updated_at
```
Comment (solnic) [inline spec/lib/generators/form_generator_spec.rb]: `timestamps` helper w/o args defaults to those, so we can simply generate a call to this helper w/o args :)

**Comment by solnic** [comment]
Passing in `attributes` is actually redundant since command configured with a specific params handler will use it. Unless we want to change that behavior for forms and explicitly do it in form and don't use any params handler for commands. No strong opinions here, just wondering what would be better.

**Comment by cflipse** [comment]
Oh.  That's good to know; I was beginning to muse about maybe-needing to block off access to params a little.

I do want to generate a spec file for this as well, I'm just working my way through what it should look like.

**Comment by solnic** [comment]
Thanks, that's a great start. We can tackle spec generation later, it's not a simple thing (support for rspec and minitest, decide on test structure etc.) and I believe a skeleton of a form object is already helpful even when there's no test file.

**Pull request merged**